cmake/SetupCompilerFlags.cmake

```diff
@@ -57,7 +57,7 @@ if (ENABLE_PEDANTIC_WARNINGS)
     DEFAULT  ""
     GNU "-Wpedantic"
     CLANG "-Wpedantic"
-    INTEL "-Wall -Wcheck -wd2259"
+    INTEL "-Wall -Wcheck -wd2259 -wd654"
```

Comment:
I think it might be worth adding a comment explaining what warnings we are disabling.

Reply:
Me too!  Especially since I cannot remember now :-)

Reply:
Turns out, this extra option was not needed so I took it back out.  Will make sure and add a comment if/when it comes back.